### PR TITLE
Add Flask guide

### PR DESCRIPTION
**Pull request by unicodeveloper**
CleanShot 2024-11-01 at 16 27 49@2x

### CODE REVIEW

**Comment by brody192** [comment]
Looks like you didn't update sidebar.ts

**Comment by unicodeveloper** [comment]
> Looks like you didn't update sidebar.ts

Oops missed that.

File: src/docs/guides/flask.md [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+    - Click [Generate Domain](/guides/public-networking#railway-provided-domain) to create a public URL for your app.
+
+## Use a Dockerfile
+
+1. Create a `Dockerfile` in the app's root directory.
+2. Add the content below to the `Dockerfile`:
+    ```docker
+    # Use the Python 3 alpine official image
+    # https://hub.docker.com/_/python
+    FROM python:3-alpine
```
Comment (brody192) [inline src/docs/guides/flask.md:194]: We try to shy away from alpine images, they have a strange networking stack and have shown to have lower CPU performance compared to the Debian counterparts.

File: src/docs/guides/flask.md [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+
```
Comment (brody192) [inline src/docs/guides/flask.md:195]: We must add `ENV PYTHONUNBUFFERED=1` here, if this is omitted log lines that get buffered could be dropped, this is something nixpacks does by default for python.

Comment (unicodeveloper) [reply src/docs/guides/flask.md:195]: Oh, that's new knowledge.

File: src/docs/guides/flask.md (hunk 1 above)
Comment (unicodeveloper) [reply src/docs/guides/flask.md:194]: Interesting...never knew that.

**Pull request merged**